src/tools/ScenarioMeasurement/SizeOnDisk/SizeOnDisk.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Reporting;
+using System;
+using System.Collections.Generic;
+using System.CommandLine.DragonFruit;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace ScenarioMeasurement
+{
+    class SizeOnDisk
+    {
+        /// <summary>
+        /// SizeOnDisk reports breakdown information about file sizes under a directory (or set of directories)
+        /// </summary>
+        /// <param name="scenarioName">Scenario name for logging purposes</param>
+        /// <param name="dirs">One or more directories to scan.</param>
+        /// <param name="reportJsonPath">Output path for lab reporting purposes</param>
+        /// <returns></returns>
+        static int Main(string scenarioName,
+                        string[] dirs,
+                        string reportJsonPath = "")
+        {
+            if (string.IsNullOrEmpty(scenarioName) || dirs is null || dirs.Length == 0)
+            {
+                Console.WriteLine("Usage: SizeOnDisk <dir0> <dir1> <dirN>");
+                return -1;
+            }
+
+            foreach(var dir in dirs)
+            {
+                if(!Directory.Exists(dir))
+                {
+                    Console.WriteLine($"Directory {dir} does not exist");
+                    return -1;
+                }
+            }
+            var directories = new Dictionary<string, Dictionary<string, long>>();
+            foreach(var dir in dirs)
+            {
+                FindVersionNumbers(dir);
+                directories.Add(dir, GetDirSize(dir));
+            }
```

Comment:
I wanted to do all the validation before I did any actual work. I can pull it together if you think that's not the right principle.

Reply:
It looks odd to me to loop through a collection twice when nothing is changing in the first loop. Also since we are tearing down the process on any failure we don't need to worry about torn state. All that to say, I don't see anything actually wrong with this.

Reply:
Sure.